geode-core/src/main/java/org/apache/geode/internal/cache/AbstractRegionMap.java

```diff
@@ -838,24 +838,23 @@ public boolean initialImagePut(final Object key, final long lastModified, Object
                 if (result) {
                   if (oldIsTombstone) {
                     owner.unscheduleTombstone(oldRe);
-                    if (newValue != Token.TOMBSTONE) {
-                      lruEntryCreate(oldRe);
-                    } else {
-                      lruEntryUpdate(oldRe);
-                    }
                   }
                   if (newValue == Token.TOMBSTONE) {
                     if (!oldIsDestroyedOrRemoved) {
                       owner.updateSizeOnRemove(key, oldSize);
                     }
                     owner.scheduleTombstone(oldRe, entryVersion);
-                    lruEntryDestroy(oldRe);
+                    if (!oldIsTombstone) {
```

Comment:
One question I have here is, when `newValue == Token.TOMBSTONE` and `oldIsTombstone == true`, it used to have `lruEntryUpdate(oldRe)`, which is in the deleted line 844. Now, after the code change, it is not having `lruEntryUpdate(oldRe)`, when `newValue == Token.TOMBSTONE` and `oldIsTombstone == true`. Is this expected?

Reply:
Yes. That was part of the problem. The code used to do this (abbreviated):
```
if (oldIsTombstone) {
  if (newValue != Token.TOMBSTONE) {
    lruEntryCreate(oldRe);
  } else {
    lruEntryUpdate(oldRe);
  }
}
if (newValue == Token.TOMBSTONE) {
  lruEntryDestroy(oldRe);
}
```
So if `oldIsTombstone` and `newValue == Token.TOMBSTONE`, both `lruEntryUpdate and `lruEntryDestroy` were called.

And, `lruEntryCreate` \ `lruEntryUpdate` were not called at all when `newValue != Token.TOMBSTONE`.